### PR TITLE
sile: 0.12.4

### PR DESCRIPTION
**Pull request by alerque**
Upstream patch release, no changes relevant to build system...

### CODE REVIEW

**Comment by alerque** [comment]
@doronbehar Something has gone sideways with this package, perhaps you have an idea why. Up until just a few days ago I was using some older nix tooling (2.4?) locally. I was able to do things like run `nix-shell` with a sile package and get v0.12.2 working great. I now have `nix` v2.6.1 and am unable to get the same things to work. For example:

```console
$ nix run nixpkgs#sile -- --version
error: builder for '/nix/store/w4myzq4vb7a4piyhr6k4v5pxxjah7cpb-sile-0.12.2.drv' failed with exit code 1;
       last 10 log lines:
       > checking luaconf.h presence... yes
       > checking for luaconf.h... yes
       > checking for Lua header version... 5.2
       > checking if Lua header version matches 5.2... yes
       > checking if LUA_VERSION is defined... yes
       > checking for library containing exp... -lm
       > checking for library containing dlopen... -ldl
       > checking for library containing lua_load... -llua
       > checking for required Lua library cassowary... not found
       > configure: error: cannot find Lua library cassowary - install from luarocks package cassowary
       For full logs, run 'nix log /nix/store/w4myzq4vb7a4piyhr6k4v5pxxjah7cpb-sile-0.12.2.drv'.
```

(truncated because this isn't the first run attempt that shows the initial part of the build).

The same thing has started showing up in CI jobs and elsewhere. This is not directly related to this bump, but something seems to have gone wrong with the Lua dependencies and perhaps this bump will just propagate the problem.

Incidentally the Flake you setup for this project still works:

```console
$ nix run github:sile-typesetter/sile -- --version
SILE v0.12.4-f35e505-flake (Lua 5.2)
```

**Comment by alerque** [comment]
@teto Perhaps you may have an idea why the LuaRocks dependencies for this seem to be missing (the `./configure` script can't find them any more even for the version of SILE prior to this PR which used to work).

**Comment by doronbehar** [comment]
There's something wrong with the `lua` interpreter created by `lua.withPackages` - building only the `luaEnv` derivation and then running `cat ./result/bin/lua` gives:

```bash
#! /nix/store/4nmqxajzaf60yjribkgvj5j54x9yvr1r-bash-5.1-p12/bin/bash -e
export LUA_PATH=${LUA_PATH-';;'}
export LUA_CPATH=${LUA_CPATH-';;'}
exec "/nix/store/bw9qp32a5q07xla9r2gkj1lji9h9hb64-lua-5.2.4/bin/lua"  "$@"
```

Whereas upstream on the sile repo, where the nixpkgs flake input is 31ffc50c571e6683e9ecc9dbcbd4a8e9914b4497 from the 21st of September, the `lua` from the `luaEnv` derivation has:

```bash
#! /nix/store/qfb4j7w2fjjq953nd9xncz5mymj5n0kb-bash-5.1-p8/bin/bash -e
export LUA_PATH=${LUA_PATH-';;'}
export LUA_PATH=$LUA_PATH${LUA_PATH:+';'}'/nix/store/a7mrmvg20w8v5krm6wjdzxphc5d376n7-lua-5.2.4-env/share/lua/5.2/?.lua;/nix/store/a7mrmvg20w8v5krm6wjdzxphc5d376n7-lua-5.2.4-env/share/lua/5.2/?/init.lua'
export LUA_CPATH=${LUA_CPATH-';;'}
export LUA_CPATH=$LUA_CPATH${LUA_CPATH:+';'}'/nix/store/a7mrmvg20w8v5krm6wjdzxphc5d376n7-lua-5.2.4-env/lib/lua/5.2/?.so'
exec "/nix/store/j6fiysph6gy416c2g6f7v3z4askb1fcr-lua-5.2.4/bin/lua"  "$@"
```

As you can see, here the `LUA_` env vars include the values of the `luaEnv` derivation, in which `cassowary` and all other lua deps are installed.

So to summarize, this error doesn't show up locally because the nixpkgs used in `flake.lock` is outdated, and does not include the nixpkgs bug that broke the lua environments (what do you think @teto ?).

In general @alerque, you should update the nix dependencies every once in a while with `nix flake update --commit-lock-file` and then run `nix build -L` so you'd be able to catch such issues earlier - as any issue that requires changes to nixpkgs.

> (truncated because this isn't the first run attempt that shows the initial part of the build)

I think you can see a full build log with `nix log nixpkgs#sile`. Also using `-L` which is equivalent to `--print-build-logs` in `nix run` and `nix build` is something you may want to start using.

**Comment by teto** [comment]
it's possible something went wrong, I did a bug change a few months ago to increase purity. A bisect would be nice. I've also noticed that wrapLuaProgram* were broken (possibly forever). Which is annoying because many programs wrap LUA_PATH themselves instead of using wrapLuaProgram, which limits the evolution of the software (even better, we should have a hook ?).

**Comment by doronbehar** [comment]
@teto fixed somewhere the luaEnv `lua` wrapper generation, and now sile seems to compile and is functional.

**Comment by teto** [comment]
One recent good fix  is the increase in purity, i.e., lua stops looking for files in /usr/lib. 
That being said I know that the LUA_PATH hook is still kinda broken. And I think wrapLuaProgram too. I would appreciate some help in that sector, especially adding tests.

**Comment by alerque** [comment]
Thanks guys, it's really nice to have this working again. Even the SILE website can't be rebuilt without the stable & flake versions both working (since the CI job that builds the examples require them). I just re-ran the GitHub Pages job and now v0.12.4 is announced on the website. :grin:
